pkg/cliplugins/workspace/plugin/kubeconfig.go

```diff
@@ -299,9 +316,58 @@ func (o *UseWorkspaceOptions) Run(ctx context.Context) error {
 		return err
 	}
 
+	bindings, err := o.getAPIBindings(ctx, o.kcpClusterClient, newServerHost)
+	if err != nil {
+		// display the error, but don't stop the current workspace from being reported.
+		fmt.Fprintf(o.ErrOut, "error checking APIBindings: %v", err)
+	}
+	if err := findUnresolvedPermissionClaims(o.Out, bindings); err != nil {
+		// display the error, but don't stop the current workspace from being reported.
+		fmt.Fprintf(o.ErrOut, "error checking APIBindings: %v", err)
+	}
+
 	return currentWorkspace(o.Out, newServerHost, shortWorkspaceOutput(o.ShortWorkspaceOutput), workspaceType)
 }
 
+// getAPIBindings retrieves APIBindings within the workspace
+func getAPIBindings(ctx context.Context, kcpClusterClient kcpclient.ClusterInterface, host string) ([]apisv1alpha1.APIBinding, error) {
+	_, clusterName, err := pluginhelpers.ParseClusterURL(host)
+	if err != nil {
+		return nil, err
+	}
+
+	apiBindings, err := kcpClusterClient.Cluster(clusterName).ApisV1alpha1().APIBindings().List(ctx, metav1.ListOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	} else if apierrors.IsNotFound(err) {
+		// nothing to do if there aren't any
+		return nil, nil
+	}
+
+	return apiBindings.Items, nil
+}
+
+// findUnresolvedPermissionClaims finds and reports any APIBindings that do not specify permission claims matching those on the target APIExport.
+func findUnresolvedPermissionClaims(out io.Writer, apiBindings []apisv1alpha1.APIBinding) error {
+	for _, binding := range apiBindings {
+		for _, exportedClaim := range binding.Status.ExportPermissionClaims {
+			warning := true
+			for _, specClaim := range binding.Spec.PermissionClaims {
+				// If the claim matches or is acknowledged, don't issue a warning.
+				ack := (specClaim.State == apisv1alpha1.ClaimAccepted || specClaim.State == apisv1alpha1.ClaimRejected)
```

Comment:
I believe this logic is correct in that it will stop examining `AcceptablePermissionClaims` at the first acknowledged one, but I'd like a double check here.

Reply:
I think we'd want to list all unacked claims?